lib/submit_collection_updater.rb

```diff
@@ -15,13 +15,17 @@ def initialize(options = {})
     @api_url  = options[:api_url]
     @api_key = options[:api_key]
     @is_gcd_protected = options[:is_gcd_protected] || false
+    @is_dry_run = options[:is_dry_run]
   end
 
   # TODO: build up some kind of errors collection
   def update_scsb_items
-    puts "Updating the following #{@barcode_to_scsb_xml_mapping.keys.length} barcodes: #{@barcode_to_scsb_xml_mapping.keys.join(',')}"
-    @barcode_to_scsb_xml_mapping.each do |barcode, scsb_xml|
-      update_item(barcode, scsb_xml)
+    if (@is_dry_run)
+      puts "This is a dry run for development. It will not update any SCSB collection item."
+    else
+      puts "Updating the following #{@barcode_to_scsb_xml_mapping.keys.length} barcodes: #{@barcode_to_scsb_xml_mapping.keys.join(',')}"
+      @barcode_to_scsb_xml_mapping.each do |barcode, scsb_xml|
+        update_item(barcode, scsb_xml)
     end
```

Comment:
Hi @ktp242, the code needs one more `end` statement on line 29, so it will complete the `.each do` loop, while the outer `end` statements would wrap up the `if` statement and `def` statement respectively.

Reply:
Maybe @nodanaonlyzuul will give us some insight on what the following means. I ran the rspec test `bundle exec rspec spec` and the test suite returned the following message:
```
....{"level":"INFO","message":"Message body: {\"action\":\"iamsupported\"} with attributes {\"SentTimestamp\"=>\"1516906421\"} and user_attributes of {}","timestamp":"2018-01-26T13:53:41.776-0500"}
.{"level":"ERROR","message":"Not valid customer code for the barcode: 5678.","timestamp":"2018-01-26T13:53:41.779-0500"}
.Updating the following 1 barcodes: 123
sent barcode 123 to submitCollection. The response was good job
.

Deprecation Warnings:

Using `any_instance` from rspec-mocks' old `:should` syntax without explicitly enabling the syntax is deprecated. Use the new `:expect` syntax or explicitly enable `:should` instead. Called from /Users/holingpoon/devl/Github/NYPL-discovery/scsb_item_updater/spec/message_handler_spec.rb:18:in `block (2 levels) in <top (required)>'.


If you need more of the backtrace for any of these deprecations to
identify where to make the necessary changes, you can configure
`config.raise_errors_for_deprecations!`, and it will turn the
deprecation warnings into errors, giving you the full backtrace.

1 deprecation warning total

Finished in 0.07323 seconds (files took 1.09 seconds to load)
7 examples, 0 failures
```

Reply:
oooops! Thanks for this!